azure-mgmt-resources/src/test/java/com/azure/management/resources/core/InterceptorManager.java

```diff
@@ -343,4 +345,9 @@ public String popVariable() {
             return recordedData.getVariables().remove();
         }
     }
+
+    private boolean isExpectedUrl(String hostUrl, String incomingUrl) {
+        Pattern pattern = Pattern.compile("roleAssignments\\/\\w+-\\w+-\\w+-\\w+-\\w+\\?");
```

Comment:
Is it due to the resource namer hasn't catched it?

Reply:
It's not created by resource namer. This name was generated from UUID inside the impls.

Reply:
Can you link the impl here?

Reply:
Please refer [this line](https://github.com/Azure/azure-libraries-for-java/blob/vnext/azure-mgmt-graph-rbac/src/main/java/com/azure/management/graphrbac/implementation/RoleAssignmentHelper.java#L92)

Reply:
This more like a hack to me.

If this is caused by parallel, personally I would prefer just do this case non-parallel (does JUnit has flag about this), instead of pattern matching here.

If it is not about parallel, ignore me.

Reply:
@weidongxu-microsoft I think it cannot work by just non-parallel running. If it is the problem of naming during multi-thread request, it can't be resolved with the ThreadLocal namer factory.

Reply:
@xccc-msft I think if we can add a function with role assignment name to solve it first.

Reply:
I see. It is caused by code change for parallel, but even if in serial run it still have same problem as parallel run.

Alright, without fix on root we probably had to bear with it. It could be more.

Reply:
@ChenTanyi @weidongxu-microsoft The problem comes from when storing the data and writing to record json. I will revert this logic and try fixing in record side.